baseband/vdif/tests/test_vdif.py

```diff
@@ -1541,3 +1541,19 @@ def test_find_header_lost_start(self, filename, tmpdir):
             header = fc.find_header()
             assert header is not None
             assert header == h1
+
+def test_edf3_vdif_payload_size():
+    """Test payload size in VDIF EDF3."""
+    testfile = str(tmpdir.join('test.vdif'))
+    with vdif.open(SAMPLE_FILE, 'rb') as fh, vdif.open(testfile, 'wb') as fw:
```

Comment:
I would open the files as a stream, with 'rs', and then inside the context manager, base yourself of `fh.header0`, read the data and then open a stream writer and write the data with
```
data1 = fr.read()
with `vdif.open(testfile, 'ws', header0=header1, nthread=8)` as fw:
    fw.write(data1)
```